### PR TITLE
Handle Access Token Refresh

### PR DESCRIPTION
**Pull request by meierj-wayscript**
## Change description

If an access token is expired, refresh it in the environment.

## Checklist

- [x] Application changes have been tested and automated tests have been added, if applicable
- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

## Related issues

> Fixes #xxxx

### CODE REVIEW

File: src/wayscript/utils.py [hunk 1]
```diff
@@ -11,13 +12,36 @@ def get_process_execution_user_token():
...
+    return value
 
 def get_process_id():
     """Return uuid of current container execution"""
     process_id = os.environ["WS_PROCESS_ID"]
     return process_id
 
 
+def get_refresh_token():
+    refresh_token = os.environ["WAYSCRIPT_EXECUTION_USER_REFRESH_TOKEN"]
```
Comment (jeffbuswell) [inline src/wayscript/utils.py:26]: Are we already setting this?

Comment (jeffbuswell) [reply src/wayscript/utils.py:26]: I don't see it when calling `printenv`

File: src/wayscript/utils.py [hunk 2]
```diff
@@ -29,36 +58,56 @@ def __init__(self, *args, **kwargs):
...
-        subpath_template = string.Template(settings.ROUTES[subpath][route])
-        subpath = subpath_template.substitute(**template_args)
-
-        url = f"{settings.WAYSCRIPT_ORIGIN}/{subpath}"
+        subpath_template_str = settings.ROUTES[subpath][route]
+        subpath_template = string.Template(subpath_template_str)
+        full_subpath = subpath_template.substitute(**template_args) if template_args else subpath_template_str
+
+        if subpath == "auth":
+            url = f"http://host.docker.internal:3001/{full_subpath}"
```
Comment (jeffbuswell) [inline src/wayscript/utils.py]: remove before checkin?

Comment (meierj-wayscript) [reply src/wayscript/utils.py]: uppppps

Comment (meierj-wayscript) [reply src/wayscript/utils.py]: removed the bad stuff

File: src/wayscript/utils.py (hunk 1 above)
Comment (meierj-wayscript) [reply src/wayscript/utils.py:26]: no, made it more gentle.

Comment (jeffbuswell) [reply src/wayscript/utils.py:26]: I mean, I don't think the `WAYSCRIPT_EXECUTION_USER_REFRESH_TOKEN` is ever set from the server side?

Comment (meierj-wayscript) [reply src/wayscript/utils.py:26]: Oooh yes I see. I must have lost the change in a merge or rebase. Added back.

**Review by jdhayford — APPROVED** [review]

**Pull request merged**